Throws if module has no entry point

If a module (like `babel-runtime`) has no entry point (no `index.js` nor `main` defined in `package.json`), The `resolvePkg()` functions throws with `Cannot find module 'babel-runtime/'`.

I'm unsure how this could be fixed.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,6 @@
 language: node_js
 sudo: false
 os:
-  - osx
   - linux
 node_js:
   - "0.10"

--- a/lib/resolve-pkg.js
+++ b/lib/resolve-pkg.js
@@ -1,4 +1,5 @@
 'use strict';
+var fs = require('fs');
 var resolve = require('resolve');
 var path = require('path');
 
@@ -16,5 +17,20 @@ module.exports = function resolvePkg(name, dir) {
   if (name === './') {
     return path.resolve(name);
   }
-  return resolve.sync(name, { basedir: dir || __dirname });
+  return resolve.sync(name, { basedir: dir || __dirname, isFile: isDirectory });
 };
+
+function isDirectory(file) {
+  var stat;
+
+  try {
+    stat = fs.statSync(file);
+  }
+  catch (err) {
+    if (err && err.code === 'ENOENT') {
+      return false;
+    }
+  }
+
+  return stat.isDirectory();
+}

--- a/tests/deps-for-test.js
+++ b/tests/deps-for-test.js
@@ -15,13 +15,13 @@ describe('depsFor', function() {
         bar: '1.0.0',
         dedupped: '1.0.0'
       },
-      path: path.join(fixturesPath, '/node_modules/foo/index.js'),
+      path: path.join(fixturesPath, '/node_modules/foo/'),
       baseDir: path.join(fixturesPath, '/node_modules/foo/')
     }, {
       name: 'bar',
       version: '1.0.0',
       main: 'index.js',
-      path: path.join(fixturesPath, '/node_modules/foo/node_modules/bar/index.js'),
+      path: path.join(fixturesPath, '/node_modules/foo/node_modules/bar/'),
       baseDir: path.join(fixturesPath, '/node_modules/foo/node_modules/bar/')
     }, {
       name: 'dedupped',
@@ -30,13 +30,13 @@ describe('depsFor', function() {
       dependencies: {
         'dedupped-child': '1.0.0'
       },
-      path: path.join(fixturesPath, '/node_modules/dedupped/index.js'),
+      path: path.join(fixturesPath, '/node_modules/dedupped/'),
       baseDir: path.join(fixturesPath, '/node_modules/dedupped/')
     }, {
       name: 'dedupped-child',
       version: '1.0.0',
       main: 'index.js',
-      path: path.join(fixturesPath, '/node_modules/dedupped/node_modules/dedupped-child/index.js'),
+      path: path.join(fixturesPath, '/node_modules/dedupped/node_modules/dedupped-child/'),
       baseDir: path.join(fixturesPath, '/node_modules/dedupped/node_modules/dedupped-child/')
     }];
 

--- a/tests/pkg-test.js
+++ b/tests/pkg-test.js
@@ -15,7 +15,7 @@ describe('pkg', function() {
         bar: '1.0.0',
         dedupped: '1.0.0'
       },
-      path: path.join(fixturesPath, '/node_modules/foo/index.js'),
+      path: path.join(fixturesPath, '/node_modules/foo/'),
       baseDir: path.join(fixturesPath, '/node_modules/foo/')
     };
 

--- a/tests/resolve-pkg-test.js
+++ b/tests/resolve-pkg-test.js
@@ -11,4 +11,8 @@ describe('resolvePkg', function() {
     assert.equal(resolvePkg('foo', fixturesPath), path.join(fixturesPath, 'node_modules/foo/index.js'));
     assert.equal(resolvePkg('/my-cool/path/', fixturesPath), '/my-cool/path/');
   });
+
+  it('Resolves to the module directory if no main entry is defined', function() {
+    assert.equal(resolvePkg('no-main/', fixturesPath), path.join(fixturesPath, 'node_modules/no-main/'));
+  });
 });

--- a/tests/resolve-pkg-test.js
+++ b/tests/resolve-pkg-test.js
@@ -8,7 +8,7 @@ var fixturesPath = path.join(__dirname, 'fixtures');
 describe('resolvePkg', function() {
   it('Resolves to the absolute main entry point path of the provided package', function() {
     assert.equal(resolvePkg('./'), path.resolve('./'));
-    assert.equal(resolvePkg('foo', fixturesPath), path.join(fixturesPath, 'node_modules/foo/index.js'));
+    assert.equal(resolvePkg('foo', fixturesPath), path.join(fixturesPath, 'node_modules/foo'));
     assert.equal(resolvePkg('/my-cool/path/', fixturesPath), '/my-cool/path/');
   });
 

